tests/test_fetch.py

```diff
@@ -0,0 +1,101 @@
+from __future__ import annotations
+
+import itertools
+from pathlib import Path
+
+import pytest
+
+from airbase.fetch import (
+    fetch_json,
+    fetch_text,
+    fetch_to_directory,
+    fetch_to_file,
+    fetch_unique_lines,
+)
```

Comment:
Can you add a couple of test cases for the fetchers when requests fail, to make sure that `raise_for_status` is respected?

Reply:
Sure

Reply:
done